sematic/resolvers/local_resolver.py

```diff
@@ -351,6 +353,7 @@ def _create_resolution(self, root_future):
         """Make a Resolution instance and persist it."""
         resolution = self._make_resolution(root_future)
         api_client.save_resolution(resolution)
+        self._resolution_was_created = True
```

Comment:
Alternatively:
```python
    self._resolution = api_client.save_resolution(resolution)
```
then test on `if self._resolution is None`.

This seems more generally useful.

Reply:
I don't want anybody to start relying on actual in-memory state of the resolution object, I would rather people query it from the server every time. Saving only the boolean here ensures that's not a risk.

Sidenote: We ideally would move in the same direction with runs/futures; basically never accessing the in-memory versions, but always querying when we need them.